core/modules/system/image.gd.inc

```diff
@@ -318,13 +318,20 @@ function image_gd_create_tmp(stdClass $image, $width, $height) {
     $transparent = imagecolortransparent($image->resource);
 
     if ($transparent >= 0) {
-      // The original must have a transparent color, allocate to the new image.
-      $transparent_color = imagecolorsforindex($image->resource, $transparent);
-      $transparent = imagecolorallocate($res, $transparent_color['red'], $transparent_color['green'], $transparent_color['blue']);
-
-      // Flood with our new transparent color.
-      imagefill($res, 0, 0, $transparent);
-      imagecolortransparent($res, $transparent);
+      if ($transparent < imagecolorstotal($image->resource)) {
+        // The original must have a transparent color, try to allocate to the new
```

Comment:
The resulting "transparent" color seems sort of random. Any color from the gif, but no transparency.

If I override $res inside this if clause `$res = imagecreate($width, $height);` - so create an indicated image instead of truecolor, the transparent background is - transparent.

Not sure, if you want to play with that.

In fact, if I replace the contents of this if clause with:

```
        $res = imagecreate($width, $height);
        imagepalettecopy($res, $image->resource);
```
the colors seem fine. Note: inside that if clause, so only when `$transparent < imagecolorstotal($image->resource)`.